core/store/src/flat/types.rs

```diff
@@ -119,6 +128,29 @@ pub enum FlatStorageCreationStatus {
     CatchingUp(CryptoHash),
 }
 
+/// This struct represents what is the current status of flat storage resharding.
+/// During resharding flat storage must be changed to reflect the new shard layout.
+///
+/// When two shards are split, the parent shard disappears and two children are created. The flat storage
+/// entries that belonged to the parent must be copied in one of the two shards. This operation happens in the
+/// background and could take significant time.
+/// After all elements have been copied the new flat storages will be behind the chain head. To remediate this issue
+/// they will enter a catching up phase. The parent shard, instead, must be removed and cleaned up.
+#[derive(
+    BorshSerialize, BorshDeserialize, Clone, Debug, PartialEq, Eq, serde::Serialize, ProtocolSchema,
+)]
+pub enum FlatStorageReshardingStatus {
+    /// Resharding phase entered when a shard is being split.
+    /// Copy key-value pairs from this shard (the parent) to children shards.
+    SplittingParent(SplittingParentStatus),
+    /// Resharding phase entered when a shard is being split.
+    /// This shard (child) is being built from state taken from its parent.
+    CreatingChild,
+    /// We apply deltas from disk until the head reaches final head.
+    /// Includes block hash of flat storage head.
+    CatchingUp(CryptoHash),
```

Comment:
It depends what we want to do

Option 1: do catchup phase for children and their flat storage will be created at chain head height or very close
Option 2: no catchup, create children at parent flat store height. This means the first block postprocessing has to go through quite a lot of deltas 

Any opinions?

Reply:
In option 2, would it happen in the main client thread? I don't think this would be a good idea.

Reply:
I think option 2 is "rate limited" in implementation so should be fine if it simplifies code.
(Note: We may need to do catchup manually in case we are looking to load memtrie from flat storage; to be checked)

Reply:
In that case I'm happy with either.